.github/workflows/verify_docker_image.yml

```diff
@@ -0,0 +1,40 @@
+name: Pull and Verify Docker Image
+
+on:
+  push:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  pull-and-build-image:
+    name: "Verify Docker Image"
+    runs-on: ubuntu-latest
```

Comment:
CPU sanity checks are okay, but we need to be testing on GPU. If testing gfx942 you'd want an AMD MI300X runner like https://github.com/ROCm/TheRock/blob/979faed95dbfb456ab9a9616258296804775231e/.github/workflows/test_linux_packages.yml#L20-L23

This is testing gfx1201 though, so we'll need an AMD RX 9070 runner

(also if using a github-hosted runner, always pin to a specific version instead  of `-latest`)